### ISSUE TITLE
Add back Socket.SendFile, Begin/EndSendFile

### ISSUE THREAD

**Issue by ericeil** [dotnet/corefx#12460]
``` csharp
    public class Socket : IDisposable {
        public IAsyncResult BeginSendFile(string fileName, AsyncCallback callback, object state);
        public IAsyncResult BeginSendFile(string fileName, byte[] preBuffer, byte[] postBuffer, TransmitFileOptions flags, AsyncCallback callback, object state);
        public void EndSendFile(IAsyncResult asyncResult);
        public void SendFile(string fileName);
        public void SendFile(string fileName, byte[] preBuffer, byte[] postBuffer, TransmitFileOptions flags);
    }
```

(Was tracked by #11788)

Additional missing APIs:
```
M:System.Net.Sockets.Socket.EndAccept(System.Byte[]@,System.IAsyncResult)
M:System.Net.Sockets.Socket.EndAccept(System.Byte[]@,System.Int32@,System.IAsyncResult)
P:System.Net.Sockets.SocketAsyncEventArgs.SendPacketsFlags
M:System.Net.Sockets.SocketAsyncEventArgs.get_SendPacketsFlags
M:System.Net.Sockets.SocketAsyncEventArgs.set_SendPacketsFlags(System.Net.Sockets.TransmitFileOptions)
```

**Comment by karelz** [comment]
Update: @geoffkizer is working on it this week.

**Comment by karelz** [comment]
@geoffkizer how far are you with the issue?

**Comment by geoffkizer** [comment]
This is mostly implemented, but I need to test on Linux.  Windows implementation is complete and passes tests.

**Comment by karelz** [comment]
Cool. BTW: You can use Linux testing in PR via CI - there is a way to kick off outerloop tests explicitly.

**Comment by karelz** [comment]
@geoffkizer this is last missing .NET Standard 2.0 API. Do you think you can finish it this week? Alternatively we can ask for help from @Priya91.
cc: @danmosemsft

**Comment by danmosemsft** [comment]
Fixed by #14158
